pkg/katautils/config.go

```diff
@@ -86,9 +86,12 @@ type factory struct {
 
 type hypervisor struct {
 	Path                    string   `toml:"path"`
+	PathList                []string `toml:"path_list"`
 	JailerPath              string   `toml:"jailer_path"`
+	JailerPathList          []string `toml:"jailer_path_list"`
 	Kernel                  string   `toml:"kernel"`
 	CtlPath                 string   `toml:"ctlpath"`
```

Comment:
> I dont think CtlPath can be configured through annotations, as far as I can tell this is a `ACRN` hypervisor specific configuration not exposed through annotations at all, just set it the configuration file.
> So I question if we need to validate this at all since, given only the root user can change the configuration.

Apparently the `CtlPath` annotation exists in the code:

```
	// CtlPath is a sandbox annotation for passing a per container path pointing at the acrn ctl binary
	CtlPath = kataAnnotHypervisorPrefix + "ctlpath"
```

I have not checked if this can be modified successfully through annotations today. In my opinion, it does not matter: if the annotation exists and can't be used, someone someday will consider it a bug worth fixing, and I'd rather have the correct template for them to follow already in place.

Reply:
@c3d This annotation did not exist before, has been added with this commit in your PR:
https://github.com/kata-containers/runtime/pull/3005/commits/999a77e56cb211d6d67c1a3a58b95ad9267b7507

Prior to your change, this value could not be changed by an annotation.
My point is we do not want to support a config as an annotation unless required to avoid bloat. Any future annotations should be added if there is a need for it and reviewed to pass validation as you have introduced here for existing annotations.

Reply:
@amshinde Can you explain the rationale why we can use annotations for any binary except ctl? This looks to me more like an oversight than a carefully designed decision.

In any case, knowing the annotation security issue, the trade-off here is between security + consistency on one hand, and bloat on the other hand. So it's interesting to evaluate the bloat. The difference in binary size on my machine is 4256 bytes, on a binary that is 36M is size. That's less than 0.01% change. It's worth it to pre-emptively close a security gap.

Reply:
@amshinde Marking as resolved, to clarify the view. Feel free to reopen if you think more action is needed here.